zonemaster/zonemaster-gui
# New ui for test results

## Purpose

Provide a better look and feel while browsing the results of a test.

## Context

#340 
https://github.com/zonemaster/zonemaster-backend/pull/1055
https://github.com/zonemaster/zonemaster-engine/pull/1144

## Changes

* Group messages by test case.
* Group test cases by modules.
* Changes severity level colors to improve contrast.
* Display test case id when test case is expanded.
* Filter buttons display a test case count instead of a message count.
* Test cases are sorted by severity, "UNSPECIFIED" have no header (always expanded) and sorted before "regular" test cases.
* Test case severity count badges.

![Screenshot 2022-11-17 at 13-46-08 ppsfleet navy · Zonemaster](https://user-images.githubusercontent.com/19394895/202450242-24f49c6a-fd8c-4060-bc9e-feae38f2956d.png)

## How to test this PR

* Install the backend and engine from https://github.com/zonemaster/zonemaster-backend/pull/1055 and https://github.com/zonemaster/zonemaster-engine/pull/1144.
* Create a new test or check the results of an old one.
